src/main/java/UtilsService.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class UtilsService {
+
+    public boolean isValidJson(String responseText) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.readTree(responseText);
+            return true;
+        } catch (Exception e) {
+            log.error("Invalid JSON response: " + responseText);
+            return false;
+        }
+    }
+
+    public String preprocessGPTResponse(String responseText) {
+        if (responseText.startsWith("```json")) {
+            responseText = responseText.substring(7).trim();
+        } else if (responseText.startsWith("```")) {
+            responseText = responseText.substring(3).trim();
+        }
+
+        if (responseText.endsWith("```")) {
+            responseText = responseText.substring(0, responseText.length() - 3).trim();
+        }
+
+        return responseText.replace("`", "").trim();
+    }
+
+    public List<String> splitTextIntoChunks(String text, int chunkSize) {
+        List<String> chunks = new ArrayList<>();
+        int length = text.length();
+        for (int i = 0; i < length; i += chunkSize) {
+            chunks.add(text.substring(i, Math.min(length, i + chunkSize)));
+        }
+        return chunks;
+    }
```

Comment:
The method is straightforward but lacks input validation. If 'text' is null or empty, it could lead to a NullPointerException. (Severity: Medium)
**Suggestion**: Add input validation to check if 'text' is null or empty before processing.